src/source/Sctp/Sctp.c

```diff
@@ -128,16 +129,28 @@ STATUS freeSctpSession(PSctpSession* ppSctpSession)
     ENTERS();
     STATUS retStatus = STATUS_SUCCESS;
     PSctpSession pSctpSession;
+    UINT64 shutdownTimeout;
 
     CHK(ppSctpSession != NULL, STATUS_NULL_ARG);
 
     pSctpSession = *ppSctpSession;
 
     CHK(pSctpSession != NULL, retStatus);
 
+    usrsctp_deregister_address(pSctpSession);
+    /* handle issue mentioned here: https://github.com/sctplab/usrsctp/issues/147
+     * the change in shutdownStatus will trigger onSctpOutboundPacket to return -1 */
+    ATOMIC_STORE(&pSctpSession->shutdownStatus, SCTP_SESSION_SHUTDOWN_INITIATED);
+
     if (pSctpSession->socket != NULL) {
+        usrsctp_set_ulpinfo(pSctpSession->socket, NULL);
+        usrsctp_shutdown (pSctpSession->socket, SHUT_RDWR);
         usrsctp_close(pSctpSession->socket);
-        usrsctp_deregister_address(pSctpSession);
+    }
+
+    shutdownTimeout = GETTIME() + DEFAULT_SCTP_SHUTDOWN_TIMEOUT;
```

Comment:
Might be cleaner with a CVAR triggering but still acceptable

Reply:
have to introduce a lock in order to use cvar. therefore sleep instead.